ark/schema/shared/traversal.ts

```diff
@@ -149,8 +154,13 @@ export class TraversalContext {
 	pathHasError(path: TraversalPath): boolean {
 		if (!this.hasError()) return false
 
-		const propString = pathToPropString(path)
-		return this.errors.some(e => propString.startsWith(e.propString))
+		let partialPropString: string = ""
+		if (Object.hasOwn(this.errors.byPath, partialPropString)) return true
```

Comment:
Can you change this to use `in` and then change `errorsByPath` to initialize to `Object.create(null)`?

You are right to consider that edge case but I think solving it in the places where we're using records is better, see https://github.com/arktypeio/arktype/issues/1121